legacy/include/distconv/tensor/channel_exchange.hpp

```diff
@@ -34,56 +34,59 @@ class ChannelExchange {
    * This does the equivalent of running a reduce-scatter on each sample in
    * src.
    */
-  virtual void reduce_scatter(TensorType &src, TensorType &dst,
-                              Al::NCCLBackend::comm_type &comm,
-                              cudaStream_t stream) {
-    // If there is only one sample, we can do this directly.
-    if (src.get_local_shape()[-1] == 1) {
-      Al::Reduce_scatter<Al::NCCLBackend, DataType>(
-        src.get_base_ptr(),
-        dst.get_base_ptr(),
-        get_sample_size(dst),
-        Al::ReductionOperator::sum,
-        comm);
-      return;
-    }
-    
-    DataType *src_buf = (DataType *) distconv::internal::RuntimeCUDA::get_device_memory_pool().get(
-      src.get_local_size()*sizeof(DataType), stream);
-    // Pack src such that we can reduce-scatter directly into dst.
-    pack_for_rs(src, dst, src_buf, comm.size(), stream);
-    Al::Reduce_scatter<Al::NCCLBackend, DataType>(
-      src_buf,
-      dst.get_base_ptr(),
-      dst.get_local_size(),
-      Al::ReductionOperator::sum,
-      comm);
-    distconv::internal::RuntimeCUDA::get_device_memory_pool().release(src_buf);
+  virtual void reduce_scatter(TensorType& src,
+                              TensorType& dst,
+                              Al::NCCLBackend::comm_type& comm,
+                              h2::gpu::DeviceStream stream)
+  {
+      // If there is only one sample, we can do this directly.
+      if (src.get_local_shape()[-1] == 1)
+      {
+          Al::Reduce_scatter<Al::NCCLBackend, DataType>(
+              src.get_base_ptr(),
+              dst.get_base_ptr(),
+              get_sample_size(dst),
+              Al::ReductionOperator::sum,
+              comm);
+          return;
+      }
+
+      DataType* src_buf =
+          (DataType*) distconv::internal::RuntimeGPU::get_device_memory_pool()
+              .get(src.get_local_size() * sizeof(DataType), stream);
+      // Pack src such that we can reduce-scatter directly into dst.
+      pack_for_rs(src, dst, src_buf, comm.size(), stream);
+      Al::Reduce_scatter<Al::NCCLBackend, DataType>(src_buf,
```

Comment:
@benson31 Should this be changed to the DNN backend, rather than NCCL?  A few lines down that type of change is done.

Reply:
I don't understand the question.

All of the Aluminum communication in this file is explicitly marked for the NCCL backend. The DNN backend (cuDNN/MIOpen) is not referenced in this file.